packages/sdk-core/src/SuperToken.ts

```diff
@@ -82,10 +83,15 @@ export default class SuperToken extends ERC20Token {
         this.idaV1 = new InstantDistributionAgreementV1({
             config: this.settings.config,
         });
-        this.underlyingToken = new ERC20Token(settings.underlyingTokenAddress);
     }
 
-    static create = async (options: ITokenOptions) => {
+    static create = async (
+        options: ITokenOptions
+    ): Promise<
+        | NativeAssetSuperToken
```

Comment:
should we pro-actively create a new type for this union type already?

Reply:
I would just use the ` SuperToken` base class here, not the union.

Reply:
`SuperTokenType`, also renamed the classes based on: https://www.notion.so/superfluidhq/Classification-of-Super-Tokens-5beace780b5c4d09a5752a3677da3dc0
*with some slight modifications for consistency:
- SuperTokenWithUnderlying => ERC20WrapperSuperToken => ERC20WrappedSuperToken
- NativeAssetSuperToken => CustomWrappedSuperToken
- SuperTokenWithoutUnderlying => CustomSuperToken

Reply:
naming addressed